src/execution/deprecated_syscall_handler.rs

```diff
@@ -32,8 +63,37 @@ impl DeprecatedOsSyscallHandlerWrapper {
             })),
         }
     }
-    pub fn call_contract(&self, syscall_ptr: Relocatable) {
-        println!("call_contract (TODO): {}", syscall_ptr);
+    pub fn call_contract(&self, syscall_ptr: Relocatable, vm: &mut VirtualMachine) -> Result<(), HintError> {
+        let sys_hand = self.deprecated_syscall_handler.as_ref().borrow();
+        let result = sys_hand
+            .exec_wrapper
+            .execution_helper
+            .as_ref()
+            .borrow_mut()
+            .result_iter
+            .next()
+            .expect("A call execution should have a corresponding result");
+
+        let response_offset = CallContract::response_offset() * 5; // TODO: response_offset() doesn't seem to take sizeof(CallContractRequest) into account
```

Comment:
Why `* 5`

Reply:
That's what the request struct's size is. I think this can be fixed now